preview/Win7Msix/Win7MSIXInstaller/Extractor.cpp

```diff
@@ -276,8 +280,22 @@ HRESULT Extractor::CreateHandler(MsixRequest * msixRequest, IPackageHandler ** i
 
 HRESULT Extractor::ExtractPackage()
 {
+    if (m_msixRequest->GetMsixResponse()->GetIsInstallCancelled())
+    {
+        return HRESULT_FROM_WIN32(ERROR_INSTALL_USEREXIT);
+    }
     RETURN_IF_FAILED(ExtractFootprintFiles());
+
+    if (m_msixRequest->GetMsixResponse()->GetIsInstallCancelled())
+    {
+        return HRESULT_FROM_WIN32(ERROR_INSTALL_USEREXIT);
+    }
     RETURN_IF_FAILED(ExtractPayloadFiles());
+
+    if (m_msixRequest->GetMsixResponse()->GetIsInstallCancelled())
+    {
+        return HRESULT_FROM_WIN32(ERROR_INSTALL_USEREXIT);
+    }

```

Comment:
Do we need these three if we are already checking in the loops within the methods called? This pattern does not look right to me and also makes the code less readable.

Reply:
I added this since Cancel could be clicked at any time, it's not needed at all places. Moved it into the loop.